src/main/java/org/hydev/mcpm/client/Downloader.java

```diff
@@ -67,4 +128,53 @@ public Downloader threads(int threads)
         this.threads = threads;
         return this;
     }
-}
+
+    class Processor implements Runnable {
```

Comment:
This is cool! Might be worth splitting this out into its own `Processor.java` file if possible, although I can see this is difficult with the `downloadFile` invocation. Your call.

Reply:
How though for "downloadFile invocation"?

Reply:
This can be probably a lambda actually:
```
executor.submit(() -> {
  String url = files.get(id);
  File to = urls.get(url);
  downloadFile(url, to);
});
```

Reply:
oooh nice. Let's me see.